addons/mrp/models/mrp_production.py

```diff
@@ -89,28 +89,22 @@ def _get_default_location_dest_id(self):
         states={'draft': [('readonly', False)]},
         help="Location where the system will stock the finished products.")
     date_planned_start = fields.Datetime(
-        'Deadline Start', copy=False, default=fields.Datetime.now,
-        index=True, required=True,
-        states={'draft': [('readonly', False)]}, oldname="date_planned")
+        'Planned Date', copy=False, default=fields.Datetime.now,
+        help="Date at which you plan to start the production.",
+        index=True, required=True, oldname="date_planned")
     date_planned_finished = fields.Datetime(
```

Comment:
what you should do

add a field date_deadline, filled by run_manufacture

**when no routing**
date_planned_start = stored
date_planned_finished = stored

add onchange on date_planned_start, set date_planned_finished = date_planned_start + 1h

**when routing**
date_planned_start + finished on the mo become readonly
when updating the date_planned_start of the FIRST workorder, update date_planned_start of the mo (override write)
when updating the date_planned_finished of the LAST workorder, update date_planned_finished of the mo (override write)